modules/@apostrophecms/migration/index.js

```diff
@@ -230,8 +230,20 @@ module.exports = {
       async migrate(options) {
         await self.apos.lock.lock(self.__meta.name);
         try {
-          for (const migration of self.migrations) {
-            await self.runOne(migration);
+          if (self.apos.isNew) {
+            // Since the site is brand new (zero documents), we may assume
+            // it requires no migrations. Mark them all as "done" but note
+            // that they were skipped, just in case we decide that's an issue later
```

Comment:
Do we have guidance for this that needs to be documented? Should all migrations be looking for skipped docs?

Reply:
No guidance is required. What's being skipped, and marked as such, is the entire migration, not an apostrophe document. The migration won't be run at all because we know it doesn't have to be if the site is brand new. We mark it just in case we change our minds about this.

Reply:
Right, it's that "just in case we change our minds" bit that I think should be documented somewhere. It could be a documentation ticket to be done later, but the feature is useless if it's not documented. If it's useless, why have it?